userspace/libsinsp/container.cpp

```diff
@@ -551,53 +553,74 @@ void sinsp_container_manager::create_engines() {
 		                                                                   m_static_name,
 		                                                                   m_static_image);
 		m_container_engines.push_back(engine);
-		m_container_engine_by_type[CT_STATIC] = engine;
+		m_container_engine_by_type[CT_STATIC].push_back(engine);
 		return;
 	}
 #if !defined(MINIMAL_BUILD) && !defined(__EMSCRIPTEN__)
 #ifndef _WIN32
 	if(m_container_engine_mask & (1 << CT_PODMAN)) {
 		auto podman_engine = std::make_shared<container_engine::podman>(*this);
 		m_container_engines.push_back(podman_engine);
-		m_container_engine_by_type[CT_PODMAN] = podman_engine;
+		m_container_engine_by_type[CT_PODMAN].push_back(podman_engine);
 	}
 	if(m_container_engine_mask & (1 << CT_DOCKER)) {
 		auto docker_engine = std::make_shared<container_engine::docker_linux>(*this);
 		m_container_engines.push_back(docker_engine);
-		m_container_engine_by_type[CT_DOCKER] = docker_engine;
+		m_container_engine_by_type[CT_DOCKER].push_back(docker_engine);
 	}
 
 	if(m_container_engine_mask & ((1 << CT_CRI) | (1 << CT_CRIO) | (1 << CT_CONTAINERD))) {
-		auto cri_engine = std::make_shared<container_engine::cri>(*this);
-		m_container_engines.push_back(cri_engine);
-		m_container_engine_by_type[CT_CRI] = cri_engine;
-		m_container_engine_by_type[CT_CRIO] = cri_engine;
-		m_container_engine_by_type[CT_CONTAINERD] = cri_engine;
+		// Get CRI socket paths from settings
```

Comment:
Is this PR still based on https://github.com/falcosecurity/libs/pull/2141?
These changes seems related to that PR. In this case, i'd move this PR as wip until the other gets merged and this one rebased.

Reply:
https://github.com/falcosecurity/libs/pull/2141 is now merged!